### PR TITLE
Improve bucket_sort_topk test coverage

### PR DESCRIPTION
**Pull request by lukatao**
Add missing accuracy test and performance benchmark for bucket_sort_topk operator.

### CODE REVIEW

File: tests/test_DSA/test_bin_topk.py [hunk 1]
```diff
@@ -349,6 +351,69 @@ def test_bucket_sort_topk_correctness():
     print(f"Overall intersection ratio: {overall_ratio:.4f}")
 
 
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA device required")
+@pytest.mark.bucket_sort_topk
+@pytest.mark.parametrize("batch_size", [1, 4, 16])
+@pytest.mark.parametrize("seq_len", [512, 1024, 4096])
+@pytest.mark.parametrize("topk", [32, 128, 512])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float16, torch.bfloat16])
+def test_bucket_sort_topk_accuracy(batch_size, seq_len, topk, dtype):
```
Comment (tengqm) [inline tests/test_DSA/test_bin_topk.py:354]:
```suggestion
def test_bucket_sort_topk(batch_size, seq_len, topk, dtype):
```

File: benchmark/test_bucket_sort_topk.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                torch.ones(batch_size, dtype=torch.int32, device=self.device) * seq_len
+            )
+            yield (inputs, starts, ends, topk)
+
+
+def _torch_topk_ref(inputs, starts, ends, topk):
+    return torch.topk(inputs, topk, dim=-1)[1].to(torch.int32)
+
+
+@pytest.mark.bucket_sort_topk
```
Comment (tengqm) [inline benchmark/test_bucket_sort_topk.py:40]: add this?

```
@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA device required")
```

File: tests/test_DSA/test_bin_topk.py [hunk 3]
```diff
@@ -349,6 +351,69 @@ def test_bucket_sort_topk_correctness():
...
+    )
+    actual_indices = bucket_sort_topk(inputs, starts, ends, topk)
+
+    assert actual_indices.shape == (batch_size, topk)
+    assert_set_similar(actual_indices, ref_indices, torch.float32)
+
+
 if __name__ == "__main__":
     # Can directly run this file for testing
     pytest.main([__file__, "-v"])
```
Comment (tengqm) [inline tests/test_DSA/test_bin_topk.py]: Remove this

File: tests/test_DSA/test_bin_topk.py (hunk 1 above)
Comment (lukatao) [reply tests/test_DSA/test_bin_topk.py:354]: Done, renamed to test_bucket_sort_topk.

File: benchmark/test_bucket_sort_topk.py (hunk 2 above)
Comment (lukatao) [reply benchmark/test_bucket_sort_topk.py:40]: Done, added the skipif decorator.

File: tests/test_DSA/test_bin_topk.py (hunk 3 above)
Comment (lukatao) [reply tests/test_DSA/test_bin_topk.py]: Done, removed.

**Comment by bin913** [comment]
@Schopenhauer-loves-Hegel Please fix failing checks

**Comment by lukatao** [comment]
The CI failure is a pre-existing Triton compilation issue in `bin_topk.py`, not caused by this PR's test changes.

The failing kernel `kernel_bucket_sort_topk` at line 102 uses `tl.cumsum(0, reverse=True)`. The `reverse` parameter for `tl.cumsum` was added in Triton 3.0, and the CI runner appears to have an older Triton version that cannot compile this specific code path with parameters `(batch_size=1, seq_len=512, topk=32, float32)`. The kernel works correctly on newer Triton versions — this PR only adds tests that happen to exercise this existing kernel path.

**Comment by lukatao** [comment]
Rechecked the latest failing GPU job: the added bfloat16 case (batch=1, seq_len=1024, topk=32) hits CUDA illegal memory access in the existing bucket_sort_topk implementation. This PR changes only tests/benchmark and still adds missing dtype/variable-length coverage. The kernel failure remains visible; no unrelated shared-kernel fix or blanket skip is being added in this PR. Evidence: https://github.com/flagos-ai/FlagGems/actions/runs/34455530605/job/102801084321
